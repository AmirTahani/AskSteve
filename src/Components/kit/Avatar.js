import React from 'react';
import { Image, StyleSheet } from 'react-native';

export default function Avatar(props) {
  return (
    <Image
      source={props.source ? props.source : require('../../assets/images/avatar.jpeg')}
      style={styles.image}
      width={50} height={50}
    />
  );
}

const styles = StyleSheet.create({
  image: {
    borderRadius: 50,
    paddingLeft: 10
  }
});
