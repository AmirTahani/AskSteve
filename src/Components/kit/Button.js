import React from 'react';
import { TouchableOpacity, StyleSheet, Text } from 'react-native';

export default function Button(props) {
  return (
    <TouchableOpacity
      style={styles.button}
      {...props}
    >
      <Text style={styles.text}>{props.children}</Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  button: {
    backgroundColor: '#6498e6',
    paddingHorizontal: 8,
    paddingVertical: 12,
    borderRadius: 8,
    marginVertical: 4
  },
  text: {
    color: '#fff',
    fontSize: 16,
    justifyContent:'center',
    alignSelf: 'center'
  }
});
