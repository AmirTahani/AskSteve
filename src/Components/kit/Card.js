import React from 'react';
import { StyleSheet, View } from 'react-native';

export default function Card(props) {
  return (
    <View style={[props.style, styles.wrapper]}>
      {props.children}
    </View>
  )
}

const styles = StyleSheet.create({
  wrapper: {
    backgroundColor: '#fff',
    elevation: 3,
    borderRadius: 8,
    marginVertical: 6,
    padding: 12,
  }
});
