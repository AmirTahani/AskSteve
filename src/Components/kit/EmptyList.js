import React from 'react';
import { Text, StyleSheet } from 'react-native';

export default function EmptyList() {
  return (
    <Text style={styles.text}>Nothing to show</Text>
  );
}

const styles = StyleSheet.create({
  text: {
    fontSize: 16,
    alignSelf: 'center',
    marginVertical: 20
  }
});
