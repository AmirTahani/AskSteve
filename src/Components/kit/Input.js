import React from 'react';
import { TextInput, StyleSheet } from 'react-native';

export default function Input(props) {
  return (
    <TextInput style={styles.input} {...props}/>
  );
}

const styles = StyleSheet.create({
  input: {
    borderRadius: 8,
    paddingHorizontal: 4,
    paddingVertical: 12,
    backgroundColor: '#fff',
    borderWidth: 1,
    borderColor: '#888888'
  }
});
