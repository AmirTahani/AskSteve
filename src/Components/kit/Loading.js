import React from 'react';
import { ActivityIndicator } from 'react-native';


export default function Loading(props) {
  return (
    <ActivityIndicator style={{marginVertical: 20}} {...props}/>
  );
}

Loading.defaultProps = {
  color: '#6498e6',
  size: 22
};
