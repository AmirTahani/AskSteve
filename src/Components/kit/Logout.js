import React from 'react';
import { Text, TouchableOpacity, StyleSheet } from "react-native";

export default function Logout(props) {
  return (
    <TouchableOpacity style={styles.button} onPress={props.onPress}>
      <Text>Logout</Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  button: {
    marginRight: 10
  }
});
