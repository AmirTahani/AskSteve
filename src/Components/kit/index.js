import Input from './Input';
import Button from './Button';
import Wrapper from './Wrapper';
import Card from './Card';
import Logout from './Logout';
import EmptyList from './EmptyList';
import Loading from './Loading';
import Avatar from './Avatar';

export {
  Input,
  Avatar,
  EmptyList,
  Loading,
  Logout,
  Button,
  Wrapper,
  Card
};
