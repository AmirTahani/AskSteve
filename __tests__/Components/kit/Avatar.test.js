import React from 'react';
import renderer from 'react-test-renderer';

import { Avatar } from '../../../src/Components/kit';


describe('<Avatar/>', () => {
  it('renders Avatar with a valid source', () => {
    const tree = renderer.create(<Avatar source={{ uri: 'https://example.com/' }}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Avatar with a default image', () => {
    const tree = renderer.create(<Avatar/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});


