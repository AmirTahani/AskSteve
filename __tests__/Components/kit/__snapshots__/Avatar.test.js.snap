// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar/> renders Avatar with a default image 1`] = `
<Image
  height={50}
  source={
    Object {
      "testUri": "../../../src/assets/images/avatar.jpeg",
    }
  }
  style={
    Object {
      "borderRadius": 50,
      "paddingLeft": 10,
    }
  }
  width={50}
/>
`;

exports[`<Avatar/> renders Avatar with a valid source 1`] = `
<Image
  height={50}
  source={
    Object {
      "uri": "https://example.com/",
    }
  }
  style={
    Object {
      "borderRadius": 50,
      "paddingLeft": 10,
    }
  }
  width={50}
/>
`;
